packages/maskbook/src/utils/hooks/useRemoteControlledDialog.ts

```diff
@@ -48,5 +55,17 @@ export function useRemoteControlledDialog<T extends { open: boolean }>(
         },
         [event, tabType, HOOK_ID],
     )
-    return [open, onUpdateByLocal] as const
+    const openDialog = useCallback(() => {
```

Comment:
```typescript
const openDialog = useCallback((ev: Omit< T>) => {
    onUpdateLocal({
        open: true,
        ...ev,
    })
}, [onUpdateLocal])
```

Reply:
In many cases, we use `openDialog`/`closeDialog` like `<Button onClick={xxxDialog} />`, if they accept an argument, the `event` object would be passed to  `onUpdateLocal` unexpectedly.

Reply:
I see, but the `ev` payload is required by `onUpdateLocal` in some cases. The original `openDialog` and `closeDialog` will lead the user to forget to set up `ev` payload.

Reply:
Right. @nuanyang233 notice same problem https://github.com/DimensionDev/Maskbook/pull/2916#discussion_r616725698. How about replace `setOpen` with `onUpdateLocal` inside `openDialog` and `closeDialog`?:

```typescript
    const openDialog = useCallback(() => {
        onUpdateByLocal({ open: true } as T)
    }, [])
    const closeDialog = useCallback(() => {
        onUpdateByLocal({ open: false } as T)
    }, [])
```

Reply:
You should infer the parameter's type of `openDialog` and `closeDialog` dynamically. If the payload type is `{ open: boolean }` then it could be void. Otherwise, It should be `Omit<typeof ev, 'open'>`.

Reply:
why not `onUpdateByLocal<T>({ open: false })`?

Reply:
May tweak later.